frontend/packages/console-shared/src/components/quick-search/QuickSearchModalBody.tsx

```diff
@@ -33,20 +37,68 @@ const QuickSearchModalBody: React.FC<QuickSearchModalBodyProps> = ({
   searchPlaceholder,
   allCatalogItemsLoaded,
   icon,
+  maxDimension,
+  viewContainer,
 }) => {
   const [catalogItems, setCatalogItems] = React.useState<CatalogItem[]>(null);
   const [catalogTypes, setCatalogTypes] = React.useState<CatalogType[]>([]);
+  const [maxHeight, setMaxHeight] = React.useState(60);
+  const [minHeight, setMinHeight] = React.useState(60);
+  const [minWidth, setMinWidth] = React.useState(60);
   const [searchTerm, setSearchTerm] = React.useState<string>(
     getQueryArgument('catalogSearch') || '',
   );
   const [selectedItemId, setSelectedItemId] = React.useState<string>('');
   const [selectedItem, setSelectedItem] = React.useState<CatalogItem>(null);
   const [viewAll, setViewAll] = React.useState<CatalogLinkData[]>(null);
-  const listCatalogItems =
-    limitItemCount > 0 ? catalogItems?.slice(0, limitItemCount) : catalogItems;
+  const [items, setItems] = React.useState<number>(limitItemCount);
+  const [modalSize, setModalSize] = React.useState<{ height: number; width: number }>();
+  const [draggableBoundary, setDraggableBoundary] = React.useState<string>(null);
   const ref = React.useRef<HTMLDivElement>(null);
   const fireTelemetryEvent = useTelemetry();
 
+  const getModalHeight = () => {
+    let height: number = 60;
+    const itemsHeight = viewAll?.length ? 388 : 365;
+    if (catalogItems?.length > 0) {
+      if (modalSize?.height >= minHeight) {
+        return modalSize?.height;
+      }
+      height += itemsHeight + (viewAll?.length - 1) * 23;
+    }
+    return height;
+  };
+
+  const listCatalogItems = React.useMemo(
+    () => (limitItemCount && limitItemCount > 0 ? catalogItems?.slice(0, items) : catalogItems),
+    [items, catalogItems, limitItemCount],
+  );
```

Comment:
Hmmm, not sure you need to memoize this... it's not doing a ton (splitting an array is relatively cheap).